Makefile

```diff
@@ -94,7 +94,14 @@ NVCC=nvcc
 OPTS=-Ofast
 LDFLAGS= -lm -pthread
 COMMON= -Iinclude/ -I3rdparty/stb/include
-CFLAGS=-Wall -Wfatal-errors -Wno-unused-result -Wno-unknown-pragmas -fPIC -rdynamic
+CFLAGS=-Wall -Wno-unused-parameter -Wno-unknown-pragmas -fPIC -rdynamic
+
+ifeq ($(USE_CPP), 1)
+# C is not C++
+CFLAGS+=-fpermissive -Wno-write-strings
```

Comment:
fpermissive really necessary?

Reply:
I guess one could  avoid it, but then one can not write idiomatic C. I would recommend to remove the USE_CPP flag anyway. What is the benefit? That C++ does better type checking is not true anymore for a long time.